mergify_engine/actions/merge_base.py

```diff
@@ -99,6 +99,64 @@ def strict_merge_parameter(v):
     raise ValueError(f"{v} is an unknown strict merge parameter")
 
 
+async def get_rule_checks_status(
+    ctxt: context.Context,
+    rule: typing.Union["rules.EvaluatedRule", "rules.EvaluatedQueueRule"],
+) -> check_api.Conclusion:
+
+    if rule.conditions.match:
+        return check_api.Conclusion.SUCCESS
+
+    conditions_without_checks = rule.conditions.copy()
+    conditions_with_all_checks = rule.conditions.copy()
+    conditions_with_check_pending = rule.conditions.copy()
+    for (
+        evaluated_condition,
+        condition_without_check,
+        condition_with_all_check,
+        condition_with_check_pending,
+    ) in zip(
+        rule.conditions.walk(),
+        conditions_without_checks.walk(),
+        conditions_with_all_checks.walk(),
+        conditions_with_check_pending.walk(),
+    ):
+        attr = evaluated_condition.get_attribute_name()
+        if attr.startswith("check-") or attr.startswith("status-"):
+            condition_without_check.update("number>0")
+            condition_with_check_pending.update_attribute_name("check-pending")
+            condition_with_all_check.update_attribute_name("check")
```

Comment:
Do we need to handle `status-` here too?

Reply:
No, this replaces all check-XXX and status-XXX in `check`.